src/use-swr.ts

```diff
@@ -165,6 +165,11 @@ async function mutate<Data = any>(
     if (shouldAbort()) return data
   }
 
+  // clear promise of revalidate whether or not `revalidate({ dedupe: false })`
+  // be called subsequently.
+  delete CONCURRENT_PROMISES[key]
+  delete CONCURRENT_PROMISES_TS[key]
+
   // enter the revalidation stage
   // update existing SWR Hooks' state
   const updaters = CACHE_REVALIDATORS[key]
```

Comment:
It might be batter if we delete the promise conditionally
```suggestion
 const updaters = CACHE_REVALIDATORS[key]
 if (updaters && updaters.length > 0) {
    const promises = []
    for (let i = 0; i < updaters.length; ++i) {
      promises.push(
        updaters[i](!!shouldRevalidate, data, error, undefined, i > 0)
      )
    }
    // return new updated value
    return Promise.all(promises).then(() => {
      if (error) throw error
      return cache.get(key)
    })
  } else {
    // clear dedupingInterval of an unmounted key
    delete CONCURRENT_PROMISES[key]
    delete CONCURRENT_PROMISES_TS[key]
  }
```

Reply:
Thanks for your suggestions.

I deleted the promise conditionally at first. I think the conditional statement should be `!!shouldRevalidate && (updaters && updaters.length > 0)` if deleting it conditionally. This conditional statement is little of complicated and hard to understand.

However, Deleting the promise after mutating is easier to understand. I would like to modify my comment to express it clearly.

Reply:
> It might be batter if we delete the promise conditionally

Thanks for your suggestions.

I deleted the promise conditionally at first. I think the conditional statement should be `!!shouldRevalidate && (updaters && updaters.length > 0)` if deleting it conditionally. This conditional statement is little of complicated and hard to understand.

However, Deleting the promise after mutating is easier to understand. I would like to modify my comment to express it clearly.